src/Symfony/Component/Yaml/Tag/TagResolver.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Yaml\Tag;
+
+use Symfony\Component\Yaml\Yaml;
+use Symfony\Component\Yaml\Exception\UnsupportedTagException;
+
+/**
+ * @author Guilhem N. <egetick@gmail.com>
+ *
+ * @internal
+ */
+final class TagResolver
```

Comment:
@nicolas-grekas imo https://github.com/symfony/symfony/pull/21194#discussion_r95410975 is a big locker in using a constructor argument. I'll submit a PR to see if we can do something about it.

> we don't need to handle the "priority" + "supports()" logic, it just adds complixity to me, no flexibility

It could be useful to allow implicit tags (`foo` becoming `!bar foo` for example, based on regexes and on its path) but I'm not sure this is very useful in userland so will try :)